app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/CommunityFragment.kt

```diff
@@ -145,7 +152,13 @@ class CommunityFragment : Fragment(R.layout.fragment_community) {
         menu.add(getString(R.string.create_new_tournament))
             .setIcon(R.drawable.ic_add)
             .setOnMenuItemClickListener {
-                replaceFragment<TournamentCreationFragment>()
+                // send args for mock to tournament creation fragment
+                val db = activity?.intent?.getStringExtra("Database")
+                val auth = activity?.intent?.getStringExtra("Auth")
+                val fragmentArgs = Bundle()
+                if (db != null) fragmentArgs.putBoolean(db, true)
+                if (auth != null) fragmentArgs.putBoolean(auth, true)
+                replaceFragment<TournamentCreationFragment>(fragmentArgs)
```

Comment:
What is this code for? It's really hard to understand.

After reading the tests I understood, but isn't it way too complicated? Why not just pass a boolean with a key? 

```
USE_MOCK_IMPLEMENTATION to true / false
USE_MOCK_FAILING_IMPLEMENTATION to true / false
```

Reply:
Well this code is the best solution I could find to pass the arguments to the tournament creation fragment, because we have 5 possible cases:
- Firebase DB + Auth
- Mock working DB + Mock working auth
- Mock working DB + Mock failing auth
- Mock failing DB + Mock working auth 
- Mock failing DB + Mock failing auth
- 
I'm adding a comment to say that it only matters when testing, otherwise `db`  and `auth` are null anyway.